condition_resource_contains_test.go

```diff
@@ -0,0 +1,58 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ladon
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestResourceContains(t *testing.T) {
+	for _, c := range []struct {
+		matches   string
+		delimiter string
+		resource  string
+		value     string
+		pass      bool
+	}{
+		//Correct matching within resource string:
+		{matches: "Exact resource component match in resource string, with : delimiter", delimiter: ":", value: "GroupId:2", resource: "resources:sensor-data:GroupId:2:deviceIds:a9b576e8-7419-4eed-a010-7f68ec0ff588", pass: true},
```

Comment:
```
	r := new(ResourceContainsCondition)
	fmt.Printf("got: %b", r.Fulfills(map[string]interface{}{"Value": "foo:bar", "Delimiter": ":"}, &Request{
		Resource: "ffoo:bar",
	}))
```

returns true which is imho a false positive

Reply:
Yes, good find, postfixing with a delimiter is not going to be enough, prefixing in a commit pushed now.